### PR TITLE
[WIP] add Gitpod config

### PR DESCRIPTION
**Pull request by marcdumais-work**
#### What it does

ATM it's difficult to produce Blueprint installers for test purposes in a
reproducable way, e.g. to test PRs. Though not mandatory for
anyone to use, Gitpod is used by several committers and permits
programmatically generating a workspace that contains the
Linux packages, e.g. Appimage, that can be started in the Gitpod
workspace (using VNC since it's an Electron app).

The Gitpod config was taken from eclipse-theia/theia and slightly
adapted for this repo. I have added the generation of preview and
production packages for Linux. This provides an easy means of testing
and looking under the hood, e.g. for troubleshooting, in an
environment that should offer reproducibility.

~~For enhanced benefit we can consider having webmaster enable the
Gitpod integration for this repo. Then we'd have automatic workspaces
pre-bbuils for PRs.~~ Already enabled at the org level I think. The Gitpod 
prebuild kicked-in as soon as a pushed this PR - see CI section below.

#### How to test

To open this PR in Gitpod and test, either use their browser
add-on for Chromium-based browsers or prepend the following URL to the GitHub URL of this PR:
"https://gitpod.io/#" .

Once merged, this trick will work with any future branch/PR/commit
in this repo, since they'll have a Gitpod config.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by thegecko — APPROVED** [review]

**Comment by marcdumais-work** [comment]
One thing I probably should have foreseen: a container is not an ideal place to run an `.AppImage` file - it doesn't work OOTB. We also can't install a `.deb` file at runtime, AFAIK (requires root/sudo). Also it's not so hot using an `Electron` app remotely with `vnc` or such - pretty much defeats the purpose of Theia supporting a browser target. 

TL;DR: I think Gitpod is not a good test environment for our Desktop packages. When we add a browser version of `Blueprint`, it will be a different thing.

For now, I think we can use Gitpod as a quick way to look-around and download pre-built Linux packages, e.g. for a given PR, to test locally. Is it worth it?
